java-frontend/src/main/java/org/sonar/java/AnalyzerMessage.java

```diff
@@ -141,6 +145,23 @@ public boolean isEmpty() {
     return location;
   }
 
+  private static AnalyzerMessage.TextSpan textSpanForTextBlock(SyntaxToken syntaxToken) {
+    String text = syntaxToken.text();
+    String[] lines = text.split("\n");
```

Comment:
I'm not sure about this. Depending of the OS, it might not be only `\n`. I would use: `"(\r)?\n|\r"`